src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java

```diff
@@ -6,272 +6,154 @@
-import com.powsybl.commons.reporter.ReporterModel;
 import com.powsybl.commons.reporter.TypedValue;
 import com.powsybl.iidm.network.Bus;
 import com.powsybl.iidm.network.Network;
-import com.powsybl.iidm.network.VariantManagerConstants;
 import com.powsybl.network.store.client.NetworkStoreService;
 import com.powsybl.network.store.client.PreloadingStrategy;
 import com.powsybl.openreac.OpenReacConfig;
 import com.powsybl.openreac.OpenReacRunner;
 import com.powsybl.openreac.parameters.input.OpenReacParameters;
 import com.powsybl.openreac.parameters.output.OpenReacResult;
 import com.powsybl.openreac.parameters.output.OpenReacStatus;
-import org.apache.commons.lang3.StringUtils;
-import org.apache.commons.lang3.tuple.Pair;
+import org.gridsuite.voltageinit.server.computation.service.AbstractResultContext;
+import org.gridsuite.voltageinit.server.computation.service.AbstractWorkerService;
+import org.gridsuite.voltageinit.server.computation.service.ExecutionService;
+import org.gridsuite.voltageinit.server.computation.service.ReportService;
+import org.gridsuite.voltageinit.server.dto.VoltageInitStatus;
 import org.gridsuite.voltageinit.server.dto.parameters.VoltageInitParametersInfos;
-import org.gridsuite.voltageinit.server.repository.VoltageInitResultRepository;
 import org.gridsuite.voltageinit.server.service.parameters.VoltageInitParametersService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
-import org.springframework.http.HttpStatus;
 import org.springframework.messaging.Message;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
 
 import java.util.*;
-import java.util.concurrent.*;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.CompletableFuture;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-import static org.gridsuite.voltageinit.server.service.NotificationService.CANCEL_MESSAGE;
-import static org.gridsuite.voltageinit.server.service.NotificationService.FAIL_MESSAGE;
+import static org.gridsuite.voltageinit.server.service.VoltageInitNotificationService.HEADER_REACTIVE_SLACKS_OVER_THRESHOLD;
+import static org.gridsuite.voltageinit.server.service.VoltageInitNotificationService.HEADER_REACTIVE_SLACKS_THRESHOLD_VALUE;
 
 /**
  * @author Etienne Homer <etienne.homer at rte-france.com>
  */
 @Service
-public class VoltageInitWorkerService {
+public class VoltageInitWorkerService extends AbstractWorkerService<OpenReacResult, VoltageInitRunContext, Void, VoltageInitResultService> {
 
+    public static final String COMPUTATION_TYPE = "VoltageInit";
     private static final Logger LOGGER = LoggerFactory.getLogger(VoltageInitWorkerService.class);
```

Comment:
This is in discussion with Tristan but I am not sure that this is needed. Isn't the Abstract class LOGGER enough ?

Reply:
I think it's nicer to have the concrete class logger in the logs

Reply:
> I think it's nicer to have the concrete class logger in the logs

yes, that what I say in the other discussion: https://github.com/gridsuite/voltage-init-server/pull/60#discussion_r1589101137